nixos/modules/config/system-path.nix

```diff
@@ -120,18 +120,17 @@ in
         "/share/vim-plugins"
       ];
 
-    environment.outputsToLink = [ "bin" "lib" "out" ];
-
     system.path = pkgs.buildEnv {
       name = "system-path";
       paths =
-        lib.filter (drv: drv != null && drv != (drv.dev or null))
-          (lib.concatMap (drv:
-            [ drv ] ++ map (outputName: drv.${outputName}.outPath or null) config.environment.outputsToLink)
-           config.environment.systemPackages);
-      inherit (config.environment) pathsToLink;
+        # The default output probably shouldn't be globally configurable.
+        # Services and users should specify them explicitly unless they want this default.
+        map (p: if p.outputUnspecified or false then p.bin or p.out or p else p)
```

Comment:
So let's use `meta.outputsToInstall` directly, and stdenv shall fill it in if not specified. The needed `nix-env` changes are clear (and trivial, I think).

I'll think on a suitable customization mechanism via the nixpkgs config, but I feel like in the long term we would better unify this perhaps through nixUP or something similar.